tasks/sast-java-sec-check.yaml

```diff
@@ -0,0 +1,71 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: sast-java-sec-check
+spec:
+  workspaces:
+    - name: workspace
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  params:
+    - name: JAR_PATH
+      description: the path of java jar file
+      type: string
+      default: "../../target/*.jar"
+    - name: OUTPUT_FORMAT
+      description: the format of findsecbugs output
+      type: string
+      default: "-sarif"
+    - name: OUTPUT_NAME
+      description: the file name of findsecbugs output
+      type: string
+      default: "sast_java_sec_output.json"
+    - name: OUTPUT_ONLYANALYZE
+      description: |
+        only analyze given classes and packages;
+        end with .* to indicate classes in a package, .- to indicate a package prefix
+      type: string
+      default: ""
+    - name: OTHER_PARAM
+      description: other parameters to run findsecbugs
+      type: string
+      default: ""
+  steps:
+    - name: compile-java
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      image: quay.io/redhat-appstudio/hacbs-test:stable
+      script: |
+        cd ../../
+        if [ -f "pom.xml" ]; then
+          mvn package
+        else
+          echo "pom.xml file doesn't exist"
+        fi
+    - name: java-sec-check
```

Comment:
I do not see reason to split the task into multiple steps, it's using one image. So three containers are spawned even when there is no pom.xml.

Reply:
So combine the three steps to one?

Reply:
I do not see reason of having three containers, especially when step is not allowed to fail.

Reply:
Ok, let me combine them.

Reply:
Done. Thanks!